sim/shaman/runes.go

```diff
@@ -95,18 +122,63 @@ func (shaman *Shaman) applyMaelstromWeapon() {
 	})
 }
 
+const ShamanPowerSurgeProcChance = .05
+
 func (shaman *Shaman) applyPowerSurge() {
 	if !shaman.HasRune(proto.ShamanRune_RuneWaistPowerSurge) {
 		return
 	}
 
+	var affectedSpells []*core.Spell
+
+	procAura := shaman.RegisterAura(core.Aura{
+		Label:    "Power Surge Proc",
+		ActionID: core.ActionID{SpellID: 440285},
+		Duration: time.Second * 10,
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+			affectedSpells = core.FilterSlice(
+				core.Flatten([][]*core.Spell{
+					shaman.ChainLightning,
+					shaman.ChainHeal,
+					{shaman.LavaBurst},
+				}), func(spell *core.Spell) bool { return spell != nil },
+			)
+		},
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			shaman.LavaBurst.CD.Reset()
+			for _, spell := range affectedSpells {
+				spell.CD.Reset()
+			}
+			core.Each(affectedSpells, func(spell *core.Spell) { spell.CastTimeMultiplier -= 1 })
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			core.Each(affectedSpells, func(spell *core.Spell) { spell.CastTimeMultiplier += 1 })
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if spell.SpellCode != int32(SpellCode_ShamanChainLightning) && spell != shaman.LavaBurst {
+				return
+			}
```

Comment:
One thing I just noticed here is this should include Chain Heal. We'd need a new SpellCode for it and to assign the spell code in `chain_heal.go`